pkg/controller/mongodb/replica_set_controller.go

```diff
@@ -334,11 +345,11 @@ func (r ReplicaSetReconciler) updateAndReturnStatusSuccess(mdb *mdbv1.MongoDB) (
 }
 
 func (r ReplicaSetReconciler) ensureAutomationConfig(mdb mdbv1.MongoDB) error {
-	cm, err := r.buildAutomationConfigConfigMap(mdb)
+	s, err := r.buildAutomationConfigSecret(mdb)
```

Comment:
:+1: Switching from a ConfigMap to a Secret for an existing cluster will reset the automation config version to 1. I checked with the agent team and they said this shouldn't cause any issues.